app/controllers/openid_application_controller.rb

```diff
@@ -94,27 +92,47 @@ def analyze_redis_launch_context
 
   def handle_opaque_token(token_string, aud)
     opaque_token = OpaqueToken.new(token_string, aud)
-    opaque_token.set_payload(fetch_issued(token_string))
-
-    return opaque_token if TokenUtil.validate_token(opaque_token)
-
+    @session = Session.find(hash_token(opaque_token))
+    if @session.nil?
+      opaque_token.set_payload(fetch_issued(token_string))
```

Comment:
This method adds the `aud` to the payload. The rest of the log checks the opaque token`s payload for the aud. As a result, cached responses will not have an aud.  Also, cached responses will not validated by charon if they are supposed to.

Reply:
Great catch.  I originally confirmed it caching when using the IAM `aud` response, but since we moved that value to be stored as part of the authorization, it was getting dropped after the first issued call.

I have updated to write the /issued `aud` value into the iam response which is then cached along with a comment.